dali/python/nvidia/dali/pool.py

```diff
@@ -0,0 +1,465 @@
+            raise RuntimeError("Chunk capacity must be positive integer")
+        if start_method == 'fork' and _b.HasCudaContext():
+            raise RuntimeError(
+                "Cannot fork process when there is CUDA context bound to it. "
+                "Make sure you build pipeline before CUDA context is acquired or change Python workers "
+                "starting method from ``fork`` to ``spawn`` (see Piepline's ``py_start_method`` option). "
+                "If you are trying to build multiple pipelines that use Python workers, you will need to "
+                "call ``start_py_workers`` method on all of them before calling ``build`` method of any pipeline.")
+        mp = multiprocessing.get_context(start_method)
+        if num_workers < 1:
+            raise RuntimeError("num_workers must be a positive integer")
+        self._num_workers = num_workers
+        self._processes = []
+        self._task_pipes = []
+        self._res_pipes = []
+        self._socks = []
+        self._task_pipes_lock = threading.Lock()
+        self._from_tracker = None
+        self._to_tracker = None
+        self._tracker_thread = None
+        for i in range(self._num_workers):
+            task_r, task_w = mp.Pipe(duplex=False)
+            res_r, res_w = mp.Pipe(duplex=False)
+            sock_reader, sock_writer = socket.socketpair()
+            process = mp.Process(
+                target=worker,
+                args=(i, callbacks, prefetch_queue_depths, initial_chunk_size,
+                      task_r, res_w, sock_writer),
+            )
+            self._task_pipes.append(task_w)
+            self._res_pipes.append(res_r)
+            self._processes.append(process)
+            self._socks.append(sock_reader)
+        self._start_processes()
+
+    def get_recv_pipes(self):
+        """Return all pipes with incoming communication.
+
+        Note: One pipe is from tracking process that may send `None` in case of shutdown.
+        """
+        return self._res_pipes + [self._from_tracker]
+
+    @property
+    def num_workers(self):
+        return self._num_workers
+
+    def pids(self):
+        """Get pids of the processes started by this pool.
+        """
+        return [proc.pid for proc in self._processes]
+
+
+    @property
+    def task_pipes_lock(self):
+        return self._task_pipes_lock
+
+    def sock(self, worker_id: int):
+        return self._socks[worker_id]
+
+    def send(self, worker_id: int, scheduled_tasks: ScheduledTasks):
+        """Send a message scheduling a task to worker `worker_id`. Needs to be done while
+        holding the `ProcPool.task_pipes_lock`."""
+        self._task_pipes[worker_id].send(scheduled_tasks)
+
+    def close(self):
+        if self._tracker_thread is None:
+            return
+        try:
+            self._to_tracker.send(None)
+        except BrokenPipeError:
+            """workers already exited, tracker_thread finished its task and exited and closed the pipe"""
```

Comment:
Is the docs string equivalent to `pass`?

Reply:
Probably it is. Will turn it into a comment. (and check if this is what it does).